draft-ietf-mls-protocol.md

```diff
@@ -1103,42 +1103,60 @@ opaque key_id<0..2^16-1>;
 
 ## Parent Hash
 
-The `parent_hash` extension serves to bind a KeyPackage to all the nodes
-above it in the group's ratchet tree. This enforces the tree invariant, meaning
-that malicious members can't lie about the state of the ratchet tree when they
-send Welcome messages to new members.
+The `parent_hash` extension serves to bind a KeyPackage inserted as part of an
+`UpdatePath` during a commit to the rest of the HPKE keys in that update path.
+Moreover, for each of those keys `parent_hash` also binds the set of parties
+that were told the corresponding secret key. This helps new members joining a group
+verify the tree invariant. That is, it prevents malicious members from
+constructing an artificial ratchet tree with a node v whose HPKE secret key is
+known to the malicious member yet where the member is not assigned a leaf in the
+subtree rooted at v. (Indeed, such a ratchet tree would violate the tree
```

Comment:
```suggestion
corresponding subtree. (Indeed, such a ratchet tree would violate the tree
```

Reply:
I'm not a fan of "corresponding" subtree because its a bit imprecise since we dont define what the correspondence is. I find "subtree rooted at v" clearer. Alternatively we could define "corresponding subtree" somewhere but my preference is to not introduce new (e.g. graph theory) terminology whenever standard terms can be used to the same end.